pilot/pkg/config/kube/gateway/inferencepool_collection.go

```diff
@@ -0,0 +1,319 @@
+			if err != nil {
+				log.Errorf("failed to generate service name for InferencePool %s: %v", pool.Name, err)
+				return nil, nil
+			}
+
+			shadowSvcInfo := shadowServiceInfo{
+				key: types.NamespacedName{
+					Name:      svcName,
+					Namespace: pool.GetNamespace(),
+				},
+				selector:   make(map[string]string, len(pool.Spec.Selector)),
+				poolName:   pool.GetName(),
+				targetPort: pool.Spec.TargetPortNumber,
+				poolUID:    pool.GetUID(),
+			}
+
+			for k, v := range pool.Spec.Selector {
+				shadowSvcInfo.selector[string(k)] = string(v)
+			}
+
+			// Make sure we reconcile the service if it is manually changed
+			_ = krt.FetchOne(ctx, services, krt.FilterKey(shadowSvcInfo.key.String()))
+
+			gatewayParentsToEnsure := sets.New[types.NamespacedName]()
+			routeList := krt.Fetch(ctx, httpRoutes, krt.FilterIndex(routesByNamespace, pool.Namespace))
+			for _, r := range routeList {
+				for _, rule := range r.Spec.Rules {
+					for _, httpBackendRef := range rule.BackendRefs {
+						if httpBackendRef.BackendRef.Group == nil || httpBackendRef.BackendRef.Kind == nil {
+							continue
+						}
+						if string(*httpBackendRef.BackendRef.Group) == gvk.InferencePool.Group &&
+							string(*httpBackendRef.BackendRef.Kind) == gvk.InferencePool.Kind &&
+							string(httpBackendRef.BackendRef.Name) == pool.ObjectMeta.Name {
+							for _, p := range r.Status.Parents {
+								if supportedControllers.Contains(p.ControllerName) {
+									ns := r.Namespace
+									if p.ParentRef.Namespace != nil && *p.ParentRef.Namespace != "" {
+										ns = string(*p.ParentRef.Namespace)
+									}
+									gatewayParentsToEnsure.Insert(types.NamespacedName{Name: string(p.ParentRef.Name), Namespace: ns})
+								}
+							}
+						}
+					}
+				}
+			}
+
+			newParents := []inferencev1alpha2.PoolStatus{}
+			for gtw := range gatewayParentsToEnsure {
+				newParents = append(newParents, *poolStatusTmpl(gtw.Name, gtw.Namespace, pool.Generation))
+			}
+
+			ipoolStatus := inferencev1alpha2.InferencePoolStatus{
+				Parents: newParents,
+			}
+
+			return &ipoolStatus, &InferencePool{
+				shadowService: shadowSvcInfo,
+				extRef:        extRef,
+			}
+		}, opts.WithName("InferenceExtension")...)
+}
+
+func poolStatusTmpl(gwName, ns string, generation int64) *inferencev1alpha2.PoolStatus {
+	return &inferencev1alpha2.PoolStatus{
+		GatewayRef: corev1.ObjectReference{
+			APIVersion: gatewayv1.GroupVersion.String(),
+			Kind:       gvk.Gateway.Kind,
+			Namespace:  ns,
+			Name:       gwName,
+		},
+		Conditions: []metav1.Condition{
+			{
+				Type:               string(inferencev1alpha2.InferencePoolConditionAccepted),
+				Status:             metav1.ConditionTrue,
+				Reason:             string(inferencev1alpha2.InferencePoolReasonAccepted),
+				Message:            "Referenced by an HTTPRoute accepted by the parentRef Gateway",
+				ObservedGeneration: generation,
+				// TODO(liorlieberman): verify that this is good.
```

Comment:
Likely not; each reconcile will change it and write to the cluster. If you look at the rest of the status writing in this package you can see we have some common utility libraries that only set the time if its changed, otherwise it keeps it. I recommend the same

Reply:
Utilized `setConditions` here